progressReport/parser.py

```diff
@@ -82,27 +78,23 @@ def read_meta(name):
             level_match = re.search(r"\s*([A-Za-z-_\s]+): #([A-Za-z0-9]+)", line)
```

Comment:
This is a massive monolithic function.  I think it would be good to separate it out for readability, but this can be accrued tech debt if you think its not worth doing at this point.